ideascube/library/forms.py

```diff
@@ -34,6 +38,93 @@ class Meta:
         exclude = ['serial', 'count']
 
 
+class BookSpecimenImportForm(forms.Form):
+    source = forms.FileField(label=_('CSV File'), required=True)
+
+    def clean_source(self):
+        if not zipfile.is_zipfile(self.cleaned_data['source']):
+            raise ValidationError(_('Uploaded file is not a Zip archive'))
+
+        with zipfile.ZipFile(self.cleaned_data['source'].file) as zip:
+            for name in zip.namelist():
+                if name.endswith('.csv'):
+                    self.cleaned_data['csvfilename'] = name
+                    break
+
+            else:
+                raise ValidationError(_('Archive does not contain a CSV'))
+
+        return self.cleaned_data['source']
+
+    def save(self):
+        source = self.cleaned_data['source'].file
+        csvfilename = self.cleaned_data['csvfilename']
+        items = []
+        errors = []
+
+        with zipfile.ZipFile(source) as zip:
+            with TextIOWrapper(zip.open(csvfilename, mode='r')) as csvfile:
+                for index, row in enumerate(csv.DictReader(csvfile)):
+                    try:
+                        data = {
+                            'barcode': row['barcode'],
+                            'comments': row['comments'],
+                            'location': row['location'],
+                            'serial': row['serial'],
+                            'file': row['file'],
+                            'item': row['item'],
+                        }
+
+                    except KeyError as e:
+                        errors.append(_(
+                            'Missing column "{}" on line {}').format(
+                                e.args[0], index + 1))
+                        continue
+
+                    # Get the related book item
+                    try:
+                        title = data.pop('item')
+                        data['item'] = Book.objects.get(name=title).pk
+
+                    except Book.DoesNotExist:
```

Comment:
Wait, `name` does not have a unicity constraint? 😮 

How is that going to work then?